DSharpPlus/Net/Rest/RestClient.cs

```diff
@@ -58,21 +69,10 @@ internal RestClient
     )
     {
         this.logger = logger;
+        this.httpClient = client;
 
-        HttpClientHandler httphandler = new()
-        {
-            UseCookies = false,
-            AutomaticDecompression = DecompressionMethods.Deflate | DecompressionMethods.GZip,
-            UseProxy = proxy != null,
-            Proxy = proxy
-        };
-
-        this.httpClient = new HttpClient(httphandler)
-        {
-            BaseAddress = new Uri(Utilities.GetApiBaseUri()),
-            Timeout = timeout
-        };
-
+        this.httpClient.BaseAddress = new Uri(Utilities.GetApiBaseUri());
+        this.httpClient.Timeout = timeout;

```

Comment:
Perhaps it'd be worthwhile to go the Remora route and expose a [rest] client builder; there's occasional reason to change what the client does (e.g., adding additional polly handlers for metrics, or using the canary api for one reason or another)

Reply:
(Client builder as in `Action<HttpClient>` parameter)

Reply:
i think this is something we should look into in a separate pull request to customize specifically rest further